src/modules/timepix3/readout/DataParser.cpp

```diff
@@ -115,36 +121,34 @@ int DataParser::parse(const char *Buffer, unsigned int Size) {
       // the ESS setup, this should correspond to an EVR pulse, indicating the
       // start of a new pulse.
     } else if (ReadoutType == 6) {
-      Timepix3TDCReadout Data;
 
       // mask and offset values are defined in DataParser.h
-      Data.Type = (DataBytes & TDC_TYPE_MASK) >> TDC_TYPE_OFFSET;
-      Data.TriggerCounter =
-          (DataBytes & TDC_TRIGGERCOUNTER_MASK) >> TDC_TRIGGERCOUNTER_OFFSET;
-      Data.Timestamp = (DataBytes & TDC_TIMESTAMP_MASK) >> TDC_TIMESTAMP_OFFSET;
-      Data.Stamp = (DataBytes & TDC_STAMP_MASK) >> TDC_STAMP_OFFSET;
+      TDCDataEvent *Data = new TDCDataEvent(
```

Comment:
To reduce the number of the copy I create the object here and later I will hand over the pointer and the ownership of this object to the observer. There should be no further copy of this data.